src/iquery.cpp

```diff
@@ -0,0 +1,178 @@
+        }
+
+        GenQuery2Input input{};
+
+        if (vm.count("query_string") == 0) {
+            fmt::print(stderr, "error: Missing QUERY_STRING\n");
+            return 1;
+        }
+
+        auto query_string = vm["query_string"].as<std::string>();
+
+        // Read from stdin.
+        if ("-" == query_string) {
+            query_string.clear();
+            std::getline(std::cin, query_string);
+        }
+
+        if (query_string.empty()) {
+            fmt::print(stderr, "error: Missing QUERY_STRING\n");
+            return 1;
+        }
+
+        input.query_string = query_string.data();
+
+        std::string zone;
+        if (vm.count("zone")) {
+            zone = vm["zone"].as<std::string>();
+            input.zone = zone.data();
+        }
+
+        if (vm["sql-only"].as<bool>()) {
+            input.sql_only = 1;
+        }
+
+        irods::experimental::client_connection conn;
+        char* sql{};
+        irods::at_scope_exit free_sql{[&sql] {
+            if (sql) {
+                std::free(sql);
+            }
+        }};
+
+#if 0
+        const auto ec =
+                procApiRequest(static_cast<RcComm*>(conn),
+                           IRODS_APN_GENQUERY2,
+                           &input,
+                           nullptr,
+                           reinterpret_cast<void**>(&sql), // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+                           nullptr);
+#else
+        const auto ec = rc_genquery2(static_cast<RcComm*>(conn), &input, &sql);
+#endif
+
+        if (ec != 0) {
+            fmt::print(stderr, "error: {}\n", ec);
+            return 1;
+        }
+
+        fmt::print("{}\n", sql);
+
+        return 0;
+    }
+    catch (const irods::exception& e) {
+        fmt::print(stderr, "error: {}\n", e.client_display_what());
+    }
+    catch (const std::exception& e) {
+        fmt::print(stderr, "error: {}\n", e.what());
+    }
+
+    return 1;
+} // main
+
+auto print_usage_info() -> void
+{
+	fmt::print(R"_(iquery - Query the iRODS Catalog
+
+Usage: iquery [OPTION]... QUERY_STRING
+
+Queries the iRODS Catalog using GenQuery2.
+
+QUERY_STRING is expected to be a string matching the GenQuery2 syntax. Failing
+to meet this requirement will result in an error.
+
+If QUERY_STRING is a hyphen (-) or is empty, input is read from stdin. Input
+taken via stdin will be viewed as the QUERY_STRING to execute. For example:
+
+    echo select COLL_NAME, DATA_NAME | iquery
+
+Mandatory arguments to long options are mandatory for short options too.
+
+Options:
+  -c, --columns         List columns supported by GenQuery2.
+      --sql-only        Print the SQL generated by the parser. The generated
+                        SQL will not be executed.
+  -z, --zone=ZONE_NAME  The name of the zone to run the query against. Defaults
+                        to the local zone.
+  -h, --help            Display this help message and exit.
+)_");
+
+	char name[] = "iquery (experimental)";
```

Comment:
Should we remove `(experimental)`?

Reply:
same

i think so - it's been proven/tested in the field and is the future.

Reply:
Done